### PR TITLE
PB-636: Responsive breakpoint switcher

### PR DESCRIPTION
**Pull request by bdenham**
## Purpose of this pull request

This pull request (PR) adds a Page Builder topic for a new feature for 2.4.2: Admin viewport switchers.

## Preview (staged)

https://docs.magedevteam.com/35/user-guide/cms/page-builder-workspace.html#viewports

## Story

[PB-636: Responsive breakpoint switcher](https://github.com/magento/magento2-page-builder/issues/636)

## Magento release version

- [x] 2.4.2

## Product editions

Is this update specific to a product edition?

- [ ] Magento Open Source only
- [x] Magento Commerce only

Is this update specific to an installed feature extension

- [ ] B2B extension
- [x] Other feature set, please specify: Page Builder

whatsnew
Page Builder adds viewport preview buttons to the admin stage so users can preview how their content will look on different devices (desktop vs. mobile). For more information, see [Viewports](https://docs.magento.com/user-guide/cms/page-builder-workspace.html#viewports).

### CODE REVIEW

File: src/cms/page-builder-workspace.md [hunk 1]
```diff
@@ -42,6 +42,40 @@ _Stage with page content_
 
 {% include page-builder-save-timeout.md %}
 
+## Viewports
+
+A viewport is the visible area of a web page that a user sees. On the stage, when it is in full-screen design mode, Page Builder displays viewport buttons above the stage to show you the content as the site user will see it on the storefront.
```
Comment (leslietilling) [inline src/cms/page-builder-workspace.md:47]:
```suggestion
A _viewport_ is the visible area of a web page that a user sees. When it is in full-screen design mode, Page Builder displays viewport buttons above the stage to show you the content as the site user will see it on the storefront.
```

File: src/cms/page-builder-workspace.md [hunk 2]
```diff
@@ -42,6 +42,40 @@ _Stage with page content_
...
+
+![Viewport buttons]({% link images/images-ee/page-builder-workspace-viewport-buttons.png %}){: .zoom}
+_Viewport buttons_
+
+Page Builder also defines breakpoints to its viewports. Breakpoints define minimum and maximum widths within which certain styles are applied. Page Builder's two viewports define the following breakpoints for its content:
```
Comment (leslietilling) [inline src/cms/page-builder-workspace.md:52]:
```suggestion
Page Builder also defines breakpoints for viewports. Breakpoints define minimum and maximum widths within which certain styles are applied. The Page Builder viewports provide the following content breakpoints:
```

File: src/cms/page-builder-workspace.md [hunk 3]
```diff
@@ -42,6 +42,40 @@ _Stage with page content_
...
+
+- **Desktop breakpoint**—`min-width: 1024px`. This breakpoint applies styles defined for viewport widths that measure 1024px and wider.
+- **Mobile breakpoints**—`max-width: 768px, min-width: 640px`. These breakpoints apply styles defined for viewport widths between 768px and 640px.
+
+Page Builder viewports provide two features: **content previews** and **breakpoint settings**.
+
+### Content previews
```
Comment (leslietilling) [inline src/cms/page-builder-workspace.md:59]: Empty line after heading

```suggestion
### Content previews

```

File: src/cms/page-builder-workspace.md [hunk 4]
```diff
@@ -42,6 +42,40 @@ _Stage with page content_
...
+By default, Page Builder provides two viewport previews:
+
+- **Desktop** — Previews your content without a predefined width. Desktop-defined styles (using breakpoint minimum width of 1024px) are still applied to the page. But the Desktop viewport width is defined by settings for container content types, like Rows. Selecting the Desktop viewport shows how your content will be styled on the storefront when the browser's page width is 1024px and wider.
```
Comment (leslietilling) [inline src/cms/page-builder-workspace.md:62]:
```suggestion
- **Desktop** — Displays the content preview without a predefined width. Desktop-defined styles (using breakpoint minimum width of 1024px) are still applied to the page. But the Desktop viewport width is defined by settings for container content types, like Rows. Selecting the Desktop viewport shows how your content will be styled on the storefront when the browser page width is 1024px and wider.
```

File: src/cms/page-builder-workspace.md [hunk 5]
```diff
@@ -42,6 +42,40 @@ _Stage with page content_
...
+- **Mobile** — Previews your content at a predefined width of 768px. Unlike the Desktop viewport, the Mobile viewport does show your page content at a width of 768px, along with the styles defined for the breakpoint widths of 768px (max) and 640px (min).
```
Comment (leslietilling) [inline src/cms/page-builder-workspace.md:63]:
```suggestion
- **Mobile** — Displays the content preview at a predefined width of 768px. Unlike the Desktop viewport, the Mobile viewport does show your page content at a width of 768px, along with the styles defined for the breakpoint widths of 768px (max) and 640px (min).
```

File: src/cms/page-builder-workspace.md [hunk 6]
```diff
@@ -42,6 +42,40 @@ _Stage with page content_
...
+
+The following screenshots show Page Builder's default responsive settings for product carousels displayed for these viewports:
```
Comment (leslietilling) [inline src/cms/page-builder-workspace.md:65]:
```suggestion
The following examples show the Page Builder default responsive settings for product carousels displayed for these viewports:
```

File: src/cms/page-builder-workspace.md [hunk 7]
```diff
@@ -42,6 +42,40 @@ _Stage with page content_
...
+
+![Desktop viewport preview]({% link images/images-ee/page-builder-workspace-viewport-desktop.png %}){: .zoom}
+_Desktop viewport preview, minimum width: 1024px_
+
+![Mobile viewport preview]({% link images/images-ee/page-builder-workspace-viewport-mobile.png %}){: .zoom}
+_Mobile viewport preview, width: 768px_
+
+### Breakpoint settings
```
Comment (leslietilling) [inline src/cms/page-builder-workspace.md:73]: Empty line after heading

```suggestion
### Breakpoint settings

```

File: src/cms/page-builder-workspace.md [hunk 8]
```diff
@@ -42,6 +42,40 @@ _Stage with page content_
...
+The viewport buttons also provide the option to apply different breakpoint styles to content types based on the selected viewport. By default, Page Builder provides breakpoint settings for the Minimum Height fields of Rows, Columns, Tabs, Tab Items, Banners, Sliders, and Slides. When you select the Mobile viewport and then open the editor for one of those content types, you can enter field values specific to the Mobile viewport. Content type fields that allow specific breakpoint settings show an icon to the right of the field, as shown here for the Row:
```
Comment (leslietilling) [inline src/cms/page-builder-workspace.md]:
```suggestion
The viewport buttons also provide the option to apply different breakpoint styles to content types based on the selected viewport. By default, Page Builder provides breakpoint settings for the _Minimum Height_ settings of Rows, Columns, Tabs, Tab Items, Banners, Sliders, and Slides. When you select the Mobile viewport and then open the editor for one of those content types, you can enter values specific to the Mobile viewport. Content type fields that allow specific breakpoint settings display an icon to the right of the field, similar to the following example for a Row:
```

**Review by leslietilling — CHANGES_REQUESTED** [review]
Some editorial updates for consistency in user guide.

File: src/cms/page-builder-workspace.md [hunk 9]
```diff
@@ -42,6 +42,43 @@ _Stage with page content_
 
 {% include page-builder-save-timeout.md %}
 
+## Viewports
+
+A viewport is the visible area of a web page that a user sees. On Page Builder's stage, when it is in Page Builder's full-screen design mode, Page Builder displays viewport buttons above the stage to show you the content as the site user will see it on the storefront.
```
[the remark this answers is not in the document]
Comment (bdenham) [reply src/cms/page-builder-workspace.md]: Yes, viewport is a standard term.

**Review by omiroshnichenko — APPROVED** [review]

**Review by leslietilling — APPROVED** [review]

**Comment by leslietilling** [comment]
Please squash when merging and add `whatsnew`

**Comment by bdenham** [comment]
running tests

**Comment by bdenham** [comment]
running tests

**Pull request merged**